docs/reference.md

```diff
@@ -260,48 +260,103 @@ The following packages are available to be used in the Node.js 6.9.1 environment
 
 ## Python actions
 
-Python actions are executed by default using Python 2.7.12.
-In addition to the standard Python library, the following packages are also available for use by Python actions.
+Python actions are executed by default using Python 3.6.0.
+In addition to the Python 3 standard library, the following packages are also available for use by Python actions.
+
+- aiohttp v1.3.3
+- appdirs v1.4.3
+- asn1crypto v0.21.1
+- async-timeout v1.2.0
+- attrs v16.3.0
+- beautifulsoup4 v4.5.1
+- cffi v1.9.1
+- chardet v2.3.0
+- click v6.7
+- cryptography v1.8.1
+- cssselect v1.0.1
+- Flask v0.12
+- gevent v1.2.1
+- greenlet v0.4.12
+- httplib2 v0.9.2
+- idna v2.5
+- itsdangerous v0.24
+- Jinja2 v2.9.5
+- kafka-python v1.3.1
+- lxml v3.6.4
+- MarkupSafe v1.0
+- multidict v2.1.4
+- packaging v16.8
+- parsel v1.1.0
+- pyasn1 v0.2.3
+- pyasn1-modules v0.0.8
+- pycparser v2.17
+- PyDispatcher v2.0.5
+- pyOpenSSL v16.2.0
+- pyparsing v2.2.0
+- python-dateutil v2.5.3
+- queuelib v1.4.2
+- requests v2.11.1
+- Scrapy v1.1.2
+- service-identity v16.0.0
+- simplejson v3.8.2
+- six v1.10.0
+- Twisted v16.4.0
+- w3lib v1.17.0
+- Werkzeug v0.12
+- yarl v0.9.8
+- zope.interface v4.3.3
+
+## Python 2 actions
+
+For a **limited time** support will continue for Python 2 actions executed using Python 2.7.12.
+In addition to the Python 2.7 standard library, the following packages are also available for use by Python actions.
+
+**Note**: Python 2 is deprecated in Open Whisk, please migrate all your Python actions to use Python 3 version 3.6+.
 
-- attrs v16.1.0
+- appdirs v1.4.3
```

Comment:
In core/python2Action/Dockerfile we pin the same packages to the same versions as the **old version** of core/pythonAction/Dockerfile.  Each of those packages may in turn import its own dependencies to form the full dependency graph.  Sometimes their setup.py or requirements.txt files specifies an exact version, sometimes a version range, and sometimes does not specify a version at all.  When the same module is requested by multiple parents, tools like pip and setuptools try attempt to do dependency negotiation to reach a compromise on what version is imported.
All of this means that not all of the packages in the dependency graph are pinned.  This in turn means that the dockerskeleton that I built one month ago might not have the same version of all packages as the dockerskeleton that I build today even if they were both built using the exact same Dockerfile.  We could chose to pin 100% of all packages in the dependency graph but that has several disadvantages.  Pip is not longer able to do dependency negotiation.  An upgrade to Jinja2 might require an upgrade to [its depandancy markupsafe](https://github.com/pallets/jinja/blob/master/setup.py#L72) but if we pin all dependencies then we would need to catch these incompatibilities manually.  Also, if Jinja dropped a dependency altogether, we might keep that pinned dependency in the build even though it is never used.

All of the above means that the package version numbers in `docs/reference.md` are just a point-in-time statement, not a guarantee.  Allowing pip & setuptools to dependency negotiation has great advantages.